packages/core/src/mls/MLSService/stores/pendingProposalsStore/pendingProposalsStore.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {CommonMLS} from '../../../types';
+
+export type PendingProposalsParams = {
+  firingDate: number;
+} & CommonMLS;
+
+interface PendingProposalsStore {
+  [groupId: string]: PendingProposalsParams;
+}
+
+const storageKey = 'pendingProposals';
+
+const getUpdateDatesMap = (): PendingProposalsStore => {
+  const storedState = localStorage.getItem(storageKey);
+  if (!storedState) {
+    return {};
+  }
+  return JSON.parse(storedState);
+};
+
+const getAllItems = (): PendingProposalsParams[] => {
+  const storedStateMap = getUpdateDatesMap();
+  return Object.values(storedStateMap);
+};
+
+const getItem = ({groupId}: CommonMLS): PendingProposalsParams | undefined => {
+  const storedState = getUpdateDatesMap();
+  return storedState[groupId];
+};
```

Comment:
Seems to be only used in tests, can probably be removed if there is not real life scenario where it's used